code/controllers/subsystem/daynightcycle.dm

```diff
@@ -94,7 +92,7 @@ Basically, you are going to overwrite the flags.
 			return
 
 /datum/subsystem/daynightcycle/proc/get_turflist()
-	for(var/turf/T in block(locate(1, 1, daynight_z_lvl), locate(world.maxx, world.maxy, daynight_z_lvl)))
```

Comment:
it may not be the case that you want daynight to run on the station z, for example if there was a map where you had a station in space but people can visit a planet surface

Reply:
Fixed